application/client/src/app/ui/layout/workspace/no-tabs-content/component.ts

```diff
@@ -53,32 +54,47 @@ export class LayoutWorkspaceNoContent {
             .then((files: File[]) => {
                 if (files.length === 0) {
                     return;
+                } else if (files.length === 1) {
+                    files.forEach((file: File) => {
+                        switch (file.type) {
+                            case FileType.Any:
+                            case FileType.Text:
+                                this.ilc()
+                                    .services.system.opener.file(file)
+                                    .text()
+                                    .catch((err: Error) => {
+                                        this.log().error(
+                                            `Fail to open text file; error: ${err.message}`,
+                                        );
+                                    });
+                                break;
+                            case FileType.Dlt:
+                                this.ilc()
+                                    .services.system.opener.file(file)
+                                    .dlt()
+                                    .catch((err: Error) => {
+                                        this.log().error(
+                                            `Fail to open dlt file; error: ${err.message}`,
+                                        );
+                                    });
+                                break;
+                        }
+                    });
+                } else {
+                    this.ilc()
+                        .services.system.session.getTabsService()
+                        .add({
+                            name: 'Multiple Files',
+                            active: true,
+                            closable: true,
+                            content: {
+                                factory: TabSourceMultipleFiles,
+                                inputs: {
+                                    files: files,
+                                },
```

Comment:
This is just a note. Nothing to change/fix. 
It's become so complicated to be in component, probably we should move it right into service. We can discuss this question on a call.

Reply:
In general. I think as a minimum solution we also should support drag and drop. I mean the user should have a possibility to drop file(s) right into view from his file explorer.